src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -116,8 +116,21 @@ FilePath sessionLockFilePath(const FilePath& sessionDir)
    return sessionDir.complete("lock_file");
 }
 
-// session dir lock (initialized by attachToSourceDatabase)
-FileLock s_sessionDirLock;
+// session dir lock (lock is acquired within 'attachToSourceDatabase()')
+boost::shared_ptr<FileLock> createSessionDirLock()
+{
+#ifdef RSTUDIO_SERVER
```

Comment:
The right way to test for this is via session options rather than the preprocessor, e.g. https://github.com/rstudio/rstudio/blob/d5ab5dfa53c057a289179826a94622b715146e79/src/cpp/session/modules/SessionAgreement.cpp#L153